### PR TITLE
gnrc_rpl: new default route when old one was deleted

### PR DESCRIPTION
**Pull request by cgundogan**
Currently, when a node has two parents and the preferred parent times out, the fib entry is not updated accordingly to point to the new preferred parent until a DIO is received.
This PR adds the new default route pointing to the next preferred parent as soon as the former one gets deleted.

~~Depends on #5135~~

### CODE REVIEW

**Comment by cgundogan** [comment]
rebased to current master and changed the approach of updating the fib

**Comment by cgundogan** [comment]
how to test?
When a node has two parents, let the preferred one (the one appearing first at `rpl` output) time out by rebooting/killing it. When the node times out, (on master) the node will not enter a new default entry in the fib until a DIO is received.

File: sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c [hunk 1]
```diff
@@ -192,12 +192,30 @@ bool gnrc_rpl_parent_add_by_addr(gnrc_rpl_dodag_t *dodag, ipv6_addr_t *addr,
...
+            fib_add_entry(&gnrc_ipv6_fib_table,
+                          dodag->iface,
+                          (uint8_t *) ipv6_addr_unspecified.u8,
+                          sizeof(ipv6_addr_t),
+                          FIB_FLAG_NET_PREFIX,
+                          parent->next->addr.u8,
+                          sizeof(ipv6_addr_t),
+                          FIB_FLAG_RPL_ROUTE,
+                          (parent->next->lifetime - now) * SEC_IN_MS);
+        }
```
Comment (BytesGalore) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: else we keep the old one?

Comment (cgundogan) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: yes, for the case that this parent (to delete) is not the preferred parent, the fib default entry is not removed and should still point to the preferred parent

Comment (cgundogan) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: oops, looked at the wrong if:

if there is no `parent->next` then we basically have no parents left, so we cannot add a new fib entry

Comment (BytesGalore) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: but we could remove the removed parent from the FIB

Comment (cgundogan) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: That's done in line 200 if I understand you correctly

Comment (BytesGalore) [inline sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c]: ahh sry :eyeglasses:

**Comment by BytesGalore** [comment]
Taking the next parent from the parent set seems a correct approach to me, and the RFC6550 doesn't give a detailed procedure which parent should be chosen [1].
So ACK

[1] https://tools.ietf.org/html/rfc6550#section-8.2.2.7

**Comment by cgundogan** [comment]
Murdock is happy - GO

**Pull request merged**